Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -164,6 +164,109 @@ The following command is listing shared libraries:
+
+![Decompilation of buttonClick function](Images/Chapters/0x06c/manual_reversing_ghidra_buttonclick_decompiled.png "Decompilation of buttonClick function")
+
+Now we have followed the complete flow and have all the information about the application flow. We also concluded that the 'hidden flag' is present in a label text and in order to determine the value of the label, we need to revisit *viewDidLoad()* function, and understand what is happening in the native function identified. Analysis of the native function is discussed in [Reviewing Disassembled Native Code](#Reviewing-Disassembled-Native-Code "Reviewing Disassembled Native Code").
+
+
+##### Reviewing Disassembled Native Code
+
+Analysing disassembled native code requires a good understanding of the calling conventions and instructions used by the underlying platform. In this section we are looking in ARM64 disassembly of the native code. A good starting point to learn about ARM architecture is available at [Introduction to ARM Assembly Basics](https://azeria-labs.com/writing-arm-assembly-part-1/ "Introduction to ARM Assembly Basics") by Azeria Labs Tutorials. A quick summary of things that we will be using in this section:
+
+- In ARM64, a register is of 64 bit in size and referred to as **Xn**, where n is a number from 0 to 31. If lower (LSB) 32bits of the register is used then referred as **Wn**.
+- The input parameters to a function are passed in **X0-X7** registers.
+- The return value of the function is passed via **X0** register.
+- Load (LDR) and store (STR) instructions are used to read or write to memory from/to a register.
+- B, BL, BLX are branch instructions used for calling a function.
+
+As mentioned above as well, Objective-C code is also compiled to native binary code, but analysing C/C++ native can be more challenging. In case of Objective-C there are various symbols (especially function names) present, which makes understanding of the code easy. From above section, presence of function names like *setText*, *isEqualStrings* helped us in quickly understanding the semantics of the code. In case of C/C++ native code, if all the debug symbols are stripped, there can be very few or no symbols present to assist in analyzing the binary. 
+
+Decompilers can help us in analyzing native code, but they should be used with caution. Modern decompilers are very sophisticated and among many techniques used by them to decompile code, few are heuristics based. Heuristics based techniques might not always give correct results, one such case being, determining the number of input parameters for a given native function. Having knowledge of analysing disassembled code, assisted with decompilers can make analyzing native code less error prone.  
```

Comment:
I want to avoid the use of only in "only a few....". Only gives an indication that we know about all the techniques and we are saying with certainty. Rather i want to keep it generic, and just state, there are heuristic based techniques in decompilers. do u feel same?

Reply:
I think that's fine. Thank you. I've corrected the suggestion accordingly :)